feat(explore-suspect-attrs): Iterating on series data, sorting or dat…

Abdkhan14 opened the pull request:
…a and tooltip

`cohortsToSeriesData` takes the cohorts and assigns a value of '0' to the chart series data if a label is not present in a cohort. Example: `cohort1: {'prod': 1, 'local': 2}` and `cohort2: {'local': 3}` -----> `series_cohort_1: [{label: 'prod', value: '1'},{label: 'local', value: '2'}]` and `series_cohort_2: [{label: 'prod', value: '0'},{label: 'local', value: '3'}]`

Note: Removed confining of tooltip, it will be cut off at the boundaries. Will set appendToBody:true (which pushes the tooltip up to document.body) along with the virtualization PR, otherwise the screen freezes trying to do it for too many charts.

Commit:
feat(explore-suspect-attrs): Hiding long x-axis labels

Commit:
feat(explore-suspect-attrs): Iterating on series data, sorting or data and tooltip

Commit:
feat(explore-suspect-attrs): Iterating on series data, sorting or data and tooltip

Commit:
Revert "feat(explore-suspect-attrs): Iterating on series data, sorting or data and tooltip"

This reverts commit fa0ccaf5c832162c1f1080115fa5c8e035aa3243.

Commit:
feat(explore-suspect-attrs): Minor iteration

Commit:
feat(explore-suspect-attrs): Minor iteration

Commit:
feat(explore-suspect-attrs): Minor iteration

DominikB2014 commented on static/app/views/explore/components/suspectTags/charts.tsx:
```diff
@@ -22,25 +31,191 @@ export function Charts({rankedAttributes}: Props) {
+  [BASELINE_SERIES_NAME]: Array<{label: string; value: string}>;
+  [SELECTED_SERIES_NAME]: Array<{label: string; value: string}>;
+} {
+  const cohort1Map = new Map(cohort1.map(({label, value}) => [label, value]));
+  const cohort2Map = new Map(cohort2.map(({label, value}) => [label, value]));
```
Curious, is there a reason you used a map here instead of an object? It looks like your just using `cohort1Map` as a simple key value mapping and not taking advantage of the benefits of a map (like that it's iterable and ordered)

Abdkhan14 committed:
feat(explore-sus-attrs): Virtualizing list of charts. (#95434)

The functionality in this PR is under the flag
`performance-spans-suspect-attributes`:
- Virtualizes the list of charts
- Adds search
- Fixes tooltip (we couldn't append tooltip to `document.body` without
virtualization. It's too expensive, when we have hundreds of charts,
otherwise).

<img width="848" height="805" alt="Screenshot 2025-07-14 at 1 14 35 AM"
src="https://github.com/user-attachments/assets/74946b5e-c946-42bf-84c9-4bc406d700fe"
/>

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Commit:
feat(explore-suspect-attrs): Resolving conflicts
